tests/bluetooth/bsim_bt/compile.sh

```diff
@@ -23,6 +23,8 @@ function compile(){
   local conf_file="${conf_file:-prj.conf}"
   local cmake_args="${cmake_args:-"-DCONFIG_COVERAGE=y"}"
   local ninja_args="${ninja_args:-""}"
+  local cc_extra_opts=( -Werror )
```

Comment:
This line seems like an unnecessary extra step(?)

If you wanted to have it caller expandable, something like this should do it:
```
local cc_extra_flags="${cc_extra_flags:-""}"
local cc_flags="-Werror ${cc_extra_flags}"
...
-DCONF_FILE=${conf_file} ${cmake_args} -DCMAKE_C_FLAGS="${cc_flags}" ${app_root}/${app} \
```
And if you wanted it caller overridable, I guess something like:
```
local cc_flags="${cc_flags:-"-Werror"}"
...
-DCONF_FILE=${conf_file} ${cmake_args} -DCMAKE_C_FLAGS="${cc_flags}" ${app_root}/${app} \
```

Reply:
I really do not see any difference. Both do the same except one - my variant is easier to scale. If you insist I add your variant.

Reply:
please do